frontend/src/components/Patchnotes.jsx

```diff
@@ -207,55 +245,113 @@ const Patchnotes = ({ game }) => {
     setModalOpen(true);
   };
 
-  const renderCategories = (category, notes) => (
-    <>
-      <h2>{category} Updates</h2>
-      {Array.isArray(notes) && notes.length > 0 && notes.map((data, index) => {
-        if (!data.title && !data.name && !data.content && (!data.generalUpdates || data.generalUpdates.length === 0) && (!data.abilityUpdates || data.abilityUpdates.length === 0)) {
-          return null;
-        }
-        return (
-          <div key={index}>
-            {data.title && <h3>{data.title}</h3>}
-            {data.name && <h3>{data.name}</h3>}
-            {data.generalUpdates && data.generalUpdates.length > 0 && (
-              <div>
-                <h4>General Updates:</h4>
-                <ul>
-                  {data.generalUpdates.map((update, idx) => (
-                    <li key={idx}>{update}</li>
-                  ))}
-                </ul>
-              </div>
-            )}
-            {data.abilityUpdates && data.abilityUpdates.length > 0 && (
-              <div>
-                <h4>Ability Updates:</h4>
-                {data.abilityUpdates.map((ability, idx) => (
-                  <div key={idx}>
-                    <h5>{ability.name}</h5>
+  const filterUpdates = (updates) => {
+    if (game !== 'overwatch') return updates;
+
+    return updates.map(update => {
+      const generalUpdates = update.generalUpdates?.map(generalUpdate => ({
+        text: generalUpdate,
+        classification: classifyUpdate(generalUpdate)
+      }));
+      const abilityUpdates = update.abilityUpdates?.map(abilityUpdate => ({
+        ...abilityUpdate,
+        content: abilityUpdate.content.map(content => ({
+          text: content,
+          classification: classifyUpdate(content)
+        }))
+      }));
+
+      return {
+        ...update,
+        generalUpdates: generalUpdates?.filter(update => filter === 'all' || update.classification === filter),
+        abilityUpdates: abilityUpdates?.map(abilityUpdate => ({
+          ...abilityUpdate,
+          content: abilityUpdate.content.filter(content => filter === 'all' || content.classification === filter)
+        })).filter(abilityUpdate => abilityUpdate.content.length > 0)
+      };
+    }).filter(update => update.generalUpdates?.length > 0 || update.abilityUpdates?.length > 0 || filter === 'all');
+  };
```

Comment:
This is hard to understand for me without further context here -- can you add a comment or respond to what this is actually doing?